### PR TITLE
Add CTR mode

### PR DESCRIPTION
**Pull request by letorbi**
This PR adds the CTR operation mode to sjcl. Even though CTR has no authentication capabilities it might be useful to have a fast and simple stream-cipher for cases where authentication is not required or where authenticated encryption modes like CCM and GCM are not available (on the other side of the communication).

The PR treats CTR mode as dangerous. This means it is not part of the default configuration and even if the CTR mode is provided by sjcl it has to be activated with the following line:

    sjcl.beware["CTR mode is dangerous because it doesn't protect message integrity."]();

These are the same measurements that are also be used to protect coders from the unreflected use of the CBC mode.

### CODE REVIEW

**Comment by letorbi** [comment]
The seem to be some errors in the pull request :/ Closing it for now.

**Comment by letorbi** [comment]
I was able to fix the bugs and also added some tests to prove that CTR is working as expected. Everything should be fine now, so I'm re-opening this PR again.

Sry for any chaos I might have caused ;)

**Comment by letorbi** [comment]
I wonder why this PR has neither been merged nor commented so far. Are there any problems with this PR I did not recognize? I would be really nice to get some feedback...

**Comment by Nilos** [comment]
I'm not working/programming 24-26th of December

**Comment by letorbi** [comment]
I understand. No problem, take your time. I just wanted to be sure that my PR has been recognized and that I'll get feedback sometime.

BTW: Merry Christmas *<:}}}

File: core/ctr.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+     * @param {bitArray} [adata=[]] The authenticated data.  It must be empty.
+     * @return The decrypted data, an array of bytes.
+     * @throws {sjcl.exception.invalid} if the IV isn't exactly 128 bits, or if any adata is specified.
+     * @throws {sjcl.exception.corrupt} if if the message is corrupt.
+     */
+    decrypt: function(prf, ciphertext, iv, adata) {
+      if (adata && adata.length) {
+        throw new sjcl.exception.invalid("ctr can't authenticate data");
+      }
+      if (sjcl.bitArray.bitLength(iv) !== 128) {
```
Comment (Nilos) [inline core/ctr.js]: From my point of view you could move this check into _calculate - less duplication

**Comment by Nilos** [comment]
:+1:  Looks good!
And: happy new year!

**Comment by letorbi** [comment]
Thx! And happy new year to you, too ;)

**Pull request merged**